ytdl_nfo/__init__.py

```diff
@@ -8,22 +9,25 @@ def main():
     parser.add_argument('input', metavar='JSON_FILE', type=str, help='Json file to convert or folder to convert in')
     parser.add_argument('-e', '--extractor', help='Specify specific extractor')
     parser.add_argument('-w', '--overwrite', action="store_true", help='Overwrite existing NFO files')
+    parser.add_argument('--regex', type=str, help='Specify regex search string to match files', default=r".json$")
```

Comment:
I would suggest defaulting to `\.info\.json$` so that other non-info JSON files in the target directory do not result in errors.

In any case, you should probably escape the period.

Reply:
(I think I'd also prefer matching against the whole string instead of searching for a substring, so defaulting to `.+\.info\.json` and using the `re.match` function instead of `re.search`, or using glob matching instead of regex. But that's a much more minor thing.)

Reply:
I agree with the `\.info\.json`, I actually implemented that commit ages ago on dev but never merged it into master so thanks for catching my inexperience bugs :).

Glob is probably the better option altogether but I didn't know it existed as part of python so thanks for making me aware. Is there an advantage for using glob here?

For now I'll stick with the regex and implement your suggestions in #9. Thank you for your suggestions

Reply:
I think the main advantage of glob is just that it's simpler and more widely used for file matching. You don't have to be very computer-literate to know what `*.info.json` means, but `\.info\.json$` is going to be confusing to many users, even technical ones. (You even missed a backslash, yourself!) Regex is great for power users, but if it were my call I'd stick to the simpler and easier option unless you anticipate a specific requirement that globs could not fulfill.

Reply:
You raise a very good point, thanks for bringing it up. I agree with your viewpoint that globs will probably be easier for more users so I'll try and work that in as default. I can't think of anything as of now which globs can't fulfill so I'll look at removing the regex flag until something comes up.